File: php/test/challenge/ChallengeTest.php
<?php

namespace Challenge;

use PHPUnit\Framework\TestCase;

class ChallengeTest extends TestCase {

    /**
     * test
     */
    public function testAddProduct() {
        $customer = new Customer();
        $address = new Address();
        $order = new Order($customer, $address);

        $order->addProduct(new Product("Product 1", ProductType::DIGITAL, 10.0), 1);
        $order->addProduct(new Product("Product 2", ProductType::BOOK, 11.0), 1);
        $order->addProduct(new Product("Product 3", ProductType::PHYSICAL, 12.0), 1);

        $this->assertEquals(count($order->items()), 3);
    }

    /**
     * @test
     */
    public function testTotalAmount() {
        $customer = new Customer();
        $address = new Address();
        $order = new Order($customer, $address);

        $order->addProduct(new Product("Product 1", ProductType::DIGITAL, 10.0), 2);
        $order->addProduct(new Product("Product 2", ProductType::DIGITAL, 15.0), 2);
        $total = $order->totalAmount();

        $this->assertEquals(50.0, $total);
    }
}

File: java/src/test/java/com/creditas/challenge/ChallengeTest.java
package com.creditas.challenge;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ChallengeTest {

    @Test
    public void testAddProduct() {
        Challenge challenge = new Challenge();
        Challenge.Customer customer = challenge.new Customer();
        Challenge.Address address = challenge.new Address();
        Challenge.Order order = challenge.new Order(customer, address);

        order.addProduct(challenge.new Product("Product 1", Challenge.ProductType.DIGITAL, 10.0), 1);
        order.addProduct(challenge.new Product("Product 2", Challenge.ProductType.BOOK, 11.0), 1);
        order.addProduct(challenge.new Product("Product 3", Challenge.ProductType.PHYSICAL, 12.0), 1);

        assertEquals(order.items().size(), 3);
    }

    @Test
    public void testTotalAmount() {
        Challenge challenge = new Challenge();
        Challenge.Customer customer = challenge.new Customer();
        Challenge.Address address = challenge.new Address();
        Challenge.Order order = challenge.new Order(customer, address);

        order.addProduct(challenge.new Product("Product 1", Challenge.ProductType.DIGITAL, 10.0), 2);
        order.addProduct(challenge.new Product("Product 2", Challenge.ProductType.DIGITAL, 15.0), 2);
        double total = order.totalAmount();

        assertEquals(total, 50.0, 0.0);
    }
}
